Library/Homebrew/livecheck/options.rb

```diff
@@ -0,0 +1,105 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    # Options to modify livecheck's behavior. These primarily come from
+    # `livecheck` blocks but they can also be set by livecheck at runtime.
+    class Options
```

Comment:
What do you think of subclassing `T::Struct`? I think it would save a lot of boilerplate code. There are a few examples in the repo already ([docs](https://sorbet.org/docs/tstruct)).

Reply:
I go into more detail in my comment below but the gist of it is that `T::Struct` can lead to nice looking code but it introduces some overhead that I would prefer to avoid. The existing implementation and `T::Struct` accomplish the same thing but the latter is a bit slower and involves more memory allocations. [Interestingly, if you subclass `T::Struct` but continue to use `initialize`, there isn't as much overhead as when you use `prop` to set up variables.]

This code will be run far more than it's modified, so I optimized for performance and minimizing memory allocations.

Reply:
I'd recommend benchmarking actual livecheck workflows, not microbenchmarking allocations. Again, the `T::Struct` overhead here should be orders of magnitude less than the resources required by any code involving network requests, and I'd prefer the code be optimized for future readers/maintainers.

Reply:
For what it's worth, the reason why I sometimes benchmark livecheck code in isolation is because network request latency varies so much that it's often impossible to distinguish any signal from noise. However, I take your point that if the difference isn't noticeable when running livecheck normally, then it may not be worth it if the changes come with caveats (like worse maintainability).